### PR TITLE
WT-12319 Estimate compaction work again when expected work is done

### PR DESCRIPTION
**Pull request by etienneptl**
Over time, the file may change, especially if there are concurrent writers. It is worth re-estimating the file once the expected work is done to refresh.

### CODE REVIEW

**Comment by etienneptl** [comment]
In the `test_compact04.py` output, we can read the following:

```
# First estimation
file size 101MB
expecting to move approx. 994 more pages (27MB), 997 total, target 20MB (20971520B)

# Actual work...

# Second estimation triggered
expecting to move approx. 58 more pages (1MB), 1056 total, target 20MB (20971520B)

# End of compaction
file size 64MB
pages rewritten: 1006 (1056 expected)
```

File: src/block/block_compact.c [hunk 1]
```diff
@@ -543,12 +544,16 @@ __compact_page_skip(
...
      */
+    if (block->compact_estimated &&
+      block->compact_pages_rewritten > block->compact_pages_rewritten_expected)
+        block->compact_estimated = false;
     if (!block->compact_estimated && block->compact_pages_reviewed >= WT_THOUSAND) {
+        compact_pages_rewritten_expected_prev = block->compact_pages_rewritten_expected;
         __block_compact_estimate_remaining_work(session, block);
         /* If no potential work has been found, exit compaction. */
-        if (block->compact_pages_rewritten_expected == 0)
+        if (compact_pages_rewritten_expected_prev == block->compact_pages_rewritten_expected)
```
Comment (pmacko86) [inline src/block/block_compact.c:556]: I would keep this condition as it was. If the estimation returns the same non-zero result as last time, it just means that the re-estimation wasn't particularly useful; it is worth to finish the compaction itself, as it is already saving us up some space.

Comment (etienneptl) [reply src/block/block_compact.c]: That's very true, thank you for spotting this. Fixed in 8ddd3d396d080064a6f425a1e6e5abf6f4b0396a.

**Review by kommiharibabu — APPROVED** [review]
LGTM

**Review by Sean04 — APPROVED** [review]
lgtm

**Review by pmacko86 — APPROVED** [review]
LGTM

**Pull request merged**